server/controllers/notifications.js

```diff
@@ -0,0 +1,113 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+
+// @desc    Get all Notifications
+// @route   GET /notifications
+// @access  Private
+exports.getNotifications = asyncHandler(async (req, res) => {
+  const { id } = req.user;
+  const userNotification = await Notification.findOne({ user: id });
+  if (userNotification && userNotification.notifications.length > 0) {
+    return res.status(200).json({
+      result: userNotification.notifications.length,
+      data: userNotification.notifications,
+    });
+  } else {
+    res.status(500);
```

Comment:
```suggestion
    res.status(404);
```
I think a 404 will be more suitable here as well. Sorry for the confusion Hayder :)